lib/github/api.nit

```diff
@@ -267,6 +267,60 @@ class GithubAPI
 		if was_error then return null
 		return milestone
 	end
+
+	# Get the Github commit comment with `id`.
+	#
+	# Returns `null` if the comment cannot be found.
+	#
+	#     var api = new GithubAPI(get_github_oauth)
+	#     var repo = api.load_repo("privat/nit")
+	#     assert repo != null
+	#     var comment = api.load_commit_comment(repo, 8982707)
+	#     assert comment.user.login == "Morriar"
+	#     assert comment.body == "For testing purposes..."
+	#     assert comment.commit.sha == "7eacb86d1e24b7e72bc9ac869bf7182c0300ceca"
+	fun load_commit_comment(repo: Repo, id: Int): nullable CommitComment do
+		var comment = new CommitComment(self, repo, id)
+		comment.load_from_github
+		if was_error then return null
+		return comment
+	end
+
+	# Get the Github issue comment with `id`.
+	#
+	# Returns `null` if the comment cannot be found.
+	#
+	#     var api = new GithubAPI(get_github_oauth)
+	#     var repo = api.load_repo("privat/nit")
+	#     assert repo != null
+	#     var comment = api.load_issue_comment(repo, 6020149)
+	#     assert comment.user.login == "privat"
+	#     assert comment.created_at.to_s == "2012-05-30T20:16:54Z"
+	#     assert comment.issue.number == 10
+	fun load_issue_comment(repo: Repo, id: Int): nullable IssueComment do
+		var comment = new IssueComment(self, repo, id)
+		comment.load_from_github
+		if was_error then return null
+		return comment
+	end
+
+	# Get the Github diff comment with `id`.
+	#
+	# Returns `null` if the comment cannot be found.
+	#
+	#     var api = new GithubAPI(get_github_oauth)
+	#     var repo = api.load_repo("privat/nit")
+	#     assert repo != null
+	#     var comment = api.load_review_comment(repo, 21010363)
+	#     assert comment.path == "src/modelize/modelize_property.nit"
+	#     assert comment.original_position == 26
+	#     assert comment.pull.number == 945
+	fun load_review_comment(repo: Repo, id: Int): nullable ReviewComment do
+		var comment = new ReviewComment(self, repo, id)
+		comment.load_from_github
+		if was_error then return null
+		return comment
```

Comment:
How do you see such a factorization?

~~~
private fun do_comment_loading(comment: Comment): nullable Comment do
    comment.load_from_github
    if was_error then return null
    return comment
end
~~~

Then in the code at the 3 call sites:

~~~
var comment = new XComment(x, y, z)
if do_comment_loading(comment) == null then return null else return comment
~~~